update.go

```diff
@@ -240,6 +286,11 @@ func initUpdate() error {
 	http.HandleFunc("/update/root", nonConcurrentUpdateHandler(rootdev.Partition(rootdev.InactiveRootPartition())))
 	http.HandleFunc("/update/switch", nonConcurrentSwitchHandler(rootdev.InactiveRootPartition()))
 	http.HandleFunc("/update/testboot", nonConcurrentTestbootHandler(rootdev.InactiveRootPartition()))
+
+	for _, extraUpdateHandler := range extraUpdateHandlers {
+		http.HandleFunc(fmt.Sprintf("/update/%s", extraUpdateHandler.name), nonConcurrentLimitedUpdateHandler(extraUpdateHandler.device, extraUpdateHandler.offset, extraUpdateHandler.maxLength))
```

Comment:
I think an extra level of hierarchy would be good in the paths. How about `/update/device-specific/<name>`?